### ISSUE TITLE
MA0029 flagging incorrectly.

### ISSUE THREAD

**Issue by grosch-intl** [meziantou/Meziantou.Analyzer#719]
### Version of the Meziantou.Analyzer NuGet package

2.0.150

### Rule Identifier

MA0029

### Target Framework

net8

### C# Language version

C# 12

### Description

In the below code, the rule is telling me to merge the where statements.  That's not, to my knowledge, possible due to the second one being a predicate.

### Reproduction Steps

Minimal code:

```c#
async Task GenerateDocumentsAsync(Guid[] wanted, Expression<Func<Document, bool>> predicate, CancellationToken cancellationToken) {
    var documentByUid = await _context.Document
        .Where(x => wanted.Contains(x.Uid))
        .Where(predicate)
        .ToDictionaryAsync(x => x.Uid, cancellationToken);
```
